CONTRIBUTING.md

```diff
@@ -0,0 +1,198 @@
+- please merge via "Squash and Merge" on GitHub to maintain a clean commit history.
+- ask for a review from someone specific if you think they would be a particularly suited reviewer (possibly noting 
+  why they are suited on the PR description)
+
+## Development conventions
+BrainGlobe includes many tools and individual repositories. We are working towards adopting common standards across 
+all repositories. Currently a handful of repositories follow a common set of development conventions:
+* [`brainreg`](https://github.com/brainglobe/brainreg)
+* [`cellfinder`](https://github.com/brainglobe/cellfinder)
+* [`cellfinder-core`](https://github.com/brainglobe/cellfinder-core)
+* [`brainreg-napari`](https://github.com/brainglobe/brainreg-napari)
+* [`cellfinder-napari`](https://github.com/brainglobe/cellfinder-napari)
+* [`brainglobe-napari-io`](https://github.com/brainglobe/brainglobe-napari-io)
+
+The conventions for these repositories is as follows:
+
+### Formatting
+
+We use [Black](https://black.readthedocs.io/en/stable/), [Flake8](https://flake8.pycqa.org/en/latest/), 
+[isort](https://pycqa.github.io/isort/) and [mypy](https://mypy.readthedocs.io/en/stable/) to ensure a consistent 
+code style. You may need to run `pre-commit install` prior to working on the code to ensure that these tests are run 
+on each commit if you haven't installed as `[dev]`.
+
+To check your code before committing, please run:
+
+- for staged files
+```python
+pre-commit run
+```
+
+- for all files in the repository
+```python
+pre-commit run -a
+```
+
+### Testing
+
+We use [pytest](https://docs.pytest.org/en/latest/) for testing. Please try to ensure that all functions 
+are tested in `tests/tests/test_unit` and all workflows/command-line tools are tested in `tests/tests/test_integration`.
+
+Some tests may take a long time \(e.g. those requiring tensorflow if you don't have a GPU\). These tests should be 
+marked with `@pytest.mark.slow`, e.g.:
+
+```python
+import pytest
+@pytest.mark.slow
+def test_something_slow():
+    slow_result = run_slow_processes()
+    assert slow_result == expected_slow_thing
+```
+
+During development, these "slow" tests can be skipped by running `pytest -m "not slow"`.
+
+### GitHub actions
+
+All commits & pull requests will be build by GitHub actions 
+([e.g. cellfinder](https://github.com/brainglobe/cellfinder/actions)). To ensure there are no issues, please check 
+that it builds: `pip install .` and run all of the tests: `pytest` before committing changes.
+
+### Versioning
+
+Versioning is taken care of using [bump2version](https://github.com/c4urself/bump2version). 
+By using the `bump2version` command, new versions are incremented, a git tag is made, and when pushed to the remote 
+repository, that version will be deployed to PyPI. Usage is as follows:
+
+* Bump to new patch version \(e.g. `1.0.0` to `1.0.1-rc0`\): `bump2version patch`
```

Comment:
I might be missing something, but why the `\(` and not just `(`. Is it for the docs?

Reply:
Yep, this is from gitbook. Can be removed though.